src/screens/NationDetailsContainer/NationDetailsScreen/index.js

```diff
@@ -217,25 +119,115 @@ class NationDetailsScreen extends Component {
-          nationName={nation.nationName}/>
+          nationName={nation.nationName}
+        />
       );
     }
+    return false;
+  }
+
+  buildTabBar() {
+    const nation = openedNation(this.props);
+
+    if (this.props.isDraft) {
+      return (
+        <View style={styles.fakeBottomBar}>
+
+          <NationActionButton
+            iconSource={AssetsImage.Actions.delete}
+            title={i18n.t('screens.nations.toolbar.delete')}
+            disable={false}
+            onPress={this.props.deleteDraft}
+          />
+          <NationActionButton
+            iconSource={AssetsImage.Actions.submit}
+            title={i18n.t('screens.nations.toolbar.submit')}
+            disable={!nationIsValid(nation)}
+            onPress={this.props.submitDraft}
+          />
+        </View>
+      );
+    }
+
+    return (
+      <View style={styles.fakeBottomBar}>
+        <NationActionButton
+          iconSource={AssetsImage.Actions.chat}
+          title={i18n.t('screens.nations.toolbar.chat')}
+          disable={false}
+          onPress={this.props.openNationChat}
+        />
+        <NationActionButton
+          iconSource={AssetsImage.Actions.map}
+          title={i18n.t('screens.nations.toolbar.map')}
+          disable
+        />
+        <NationActionButton
+          iconSource={AssetsImage.Actions.join}
+          title={i18n.t('screens.nations.toolbar.join')}
+          disable={this.disableJoinButton(nation)}
+          onPress={this.props.joinNation}
+        />
+        <NationActionButton
+          iconSource={AssetsImage.Actions.leave}
+          title={i18n.t('screens.nations.toolbar.leave')}
+          disable={this.disableLeaveButton(nation)}
+          onPress={this.props.leaveNation}
+        />
+      </View>
+    );
+  }
+  render() {
+    const nation = openedNation(this.props);
+
+    if (!nation) {
+      this.props.navigator.pop();
+      return <BackgroundImage />;
+    }
+
+    const status = resolveStatus(nation);
+    const statusDescription = (status !== null ? i18n.ifExists(`screens.nationDetails.statusDescription.${status.key}`) : '');
+
+    return (
+      <View style={styles.screenContainer}>
+        <BackgroundImage />
+        <FakeNavigationBar navBarHidden='' />
+        <View style={styles.bodyContainer}>
+          <View style={styles.titleContainer}>
+            <View style={styles.titleBarLarge}>
+              <Text style={styles.largeTitle}>{nation.nationName}</Text>
+            </View>
+          </View>
+
+          <ScrollView>
+            {statusDescription !== '' && this.buildStatusPanel(statusDescription)}
+
+            {this.buildAboutView(nation)}
+            {this.buildCitizenPanel(nation)}
+            {this.buildGovernmentalStructureView(nation)}
+            {this.buildFactsView(nation)}
+          </ScrollView>
+        </View>
+        {this.buildTabBar()}
+      </View>
+    );
   }
 }
 
 NationDetailsScreen.propTypes = {
+  navigator: PropTypes.shape({ pop: {} }),
```

Comment:
This PropTypes declaration is invalid. You need to use PropTypes constant in nested object as well, like
PropTypes.shape({ pop: PropTypes.func }).
Do you mind to enable flow for that file and remove PropTypes declaration at all?

Reply:
Solved